slither/detectors/statements/msg_value_in_loop.py

```diff
@@ -0,0 +1,84 @@
+from slither.core.cfg.node import NodeType
+from slither.detectors.abstract_detector import AbstractDetector, DetectorClassification
+from slither.core.declarations import SolidityVariableComposed
+
+
+def detect_msg_value_in_loop(contract):
+    results = []
+    for f in contract.functions + contract.modifiers:
+        if f.is_implemented and f.payable:
+            msg_value_in_loop(f.entry_point, False, [], results)
+    return results
+
+
+def msg_value_in_loop(node, in_loop, visited, results):
+    if node in visited:
+        return
+    # shared visited
+    visited.append(node)
+
+    if node.type == NodeType.STARTLOOP:
+        in_loop = True
```

Comment:
The true/false might not work with nested loops. Maybe we could use a counter instead?

I think this pattern is used in other detectors, so we could improve the other detectors also